File: livingdoc-api/src/main/java/org/livingdoc/api/fixtures/scenarios/ScenarioFixture.java
package org.livingdoc.api.fixtures.scenarios;

import org.livingdoc.api.After;
import org.livingdoc.api.Before;

import java.lang.annotation.ElementType;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Annotating a class with this annotation declares it to be a fixture for a scenario example.
 * <p>
 * <b>Scenario:</b> <i>An example describing a single test case as a sequence of steps. Each step is represented as a
 * sentence containing either one or more test inputs or one or more expectations.</i>
 * <p>
 * LivingDoc will evaluate the fixture class for consistency when it is first loaded. Only scenario related annotations can
 * be used within a scenario fixture!
 *
 * @see Before
 * @see After
 * @see Step
 * @since 2.0
 */
@Repeatable(ScenarioFixture.ScenarioFixtures.class)
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface ScenarioFixture {

    String[] value();

    @Target(ElementType.TYPE)
    @Retention(RetentionPolicy.RUNTIME)
    @interface ScenarioFixtures {
        ScenarioFixture[] value();
    }

}

File: livingdoc-api/src/main/kotlin/org/livingdoc/api/fixtures/scenarios/ScenarioFixture.kt
package org.livingdoc.api.fixtures.scenarios

import org.livingdoc.jvm.api.fixture.FixtureAnnotation
import org.livingdoc.api.Before
import org.livingdoc.api.After

/**
 * Annotating a class with this annotation declares it to be a fixture for a scenario example.
 *
 * **Scenario:** *An example describing a single test case as a sequence of steps. Each step is represented as a
 * sentence containing either one or more test inputs or one or more expectations.*
 *
 * LivingDoc will evaluate the fixture class for consistency when it is first loaded. Only scenario related annotations
 * can be used within a scenario fixture!
 *
 * @see Before
 * @see After
 * @see Step
 *
 * @since 2.0
 */
@Target(AnnotationTarget.CLASS)
@FixtureAnnotation
annotation class ScenarioFixture(vararg val value: String)
